### PR TITLE
Pull Request for Issue2205: Adding option to export ICR

### PR DESCRIPTION
**Pull request by helfrij**
- Added flag to the scan configuration DB object for the 'collect ICRs' preference. The flag is false by default.
- Added getter to the XRF detector that returns the ICR controls list.
- Added methods to BioXASBeamline that handle mapping a detector to the ICRs detectors, similar to what's already implemented for the spectra controls.
- Added checkbox to the scan configuration view, allowing a user to select whether they want to collect ICRs.

### CODE REVIEW

**Comment by XuetaoLiu-CLS** [comment]
Looks good to me.

Time Estimate: 10 minutes

File: source/beamline/BioXAS/BioXASBeamline.cpp [hunk 1]
```diff
@@ -957,6 +971,83 @@ bool BioXASBeamline::clearDetectorElements(AMDetector *detector)
...
+{
+	bool result = false;
+
+	AMDetectorSet *icrs = detectorElementsMap_.value(detector, 0);
+
+	// Remove all ICRs from the detectors set, remove map entry,
+	// and delete the set.
+
+	if (icrs) {
+		removeDetectorElements(detector);
```
Comment (dretrex) [inline source/beamline/BioXAS/BioXASBeamline.cpp]: Is this supposed to be `removeDetectorICRs()`?

**Comment by dretrex** [comment]
Just the one question.
Time Estimate: 5 minutes

File: source/beamline/BioXAS/BioXASBeamline.cpp (hunk 1 above)
Comment (helfrij) [inline source/beamline/BioXAS/BioXASBeamline.cpp]: Yes, it is. Good catch!

**Pull request merged**